virtualization-host-formula/virtualization-host/_states/bootloader.py

```diff
@@ -39,15 +39,14 @@ def grub_set_default(name):
 
         # Regenerate grub config
         out = None
-        if not __opts__['test']:
-            try:
-                out = __salt__['cmd.run']('grub2-mkconfig -o /boot/grub2/grub.cfg', raise_err=True)
-            except CommandExecutionError:
-                ret['comment'] = 'Failed to run grub2-mkconfig: {0}'.format(out)
-                ret['result'] = False
-            ret['result'] = True
-        else:
+        if __opts__['test']:
             ret['result'] = None
+            return ret
+        try:
+            out = __salt__['cmd.run']('grub2-mkconfig -o /boot/grub2/grub.cfg', raise_err=True)
+        except CommandExecutionError:
+            ret['comment'] = 'Failed to run grub2-mkconfig: {0}'.format(out)
+        ret['result'] = True
     else:
```

Comment:
IIUC this `else` comes from `if len(filtered_entries) > 0`, you can avoid the indentation using the reverse check and adding the error message and the `return` in the `if` leg, avoiding the `else`

The pattern is like this:

```
if something is wrong1:
  ret['comment'] = 'wrong1'
  return ret

if something is wrong2:
  ...
  return ret

if test:
  ret['result'] = 'some changes are needed'
  ret['changes'] = ['maybe my list of possible changes']

do_changes()

if something fails:
  ..
  return ret

ret['changes'] = [...]
ret['result'] = True
return ret
```